vm/systemSmartContracts/governance.go

```diff
@@ -753,34 +753,40 @@ func (g *governanceContract) clearEndedProposals(args *vmcommon.ContractCallInpu
 			return vmcommon.UserError
 		}
 		err = g.clearUserVotes(voter, lastEndedNonce)
+		if err != nil {
+			g.eei.AddReturnMessage(err.Error())
+			return vmcommon.UserError
+		}
+
 	}
 	return vmcommon.Ok
 }
 
-func (g *governanceContract) clearUserVotes(address []byte, lastEndedNonce uint64) error {
-	userVoteList, err := g.getUserVotesV2(address)
-	if err != nil {
-		return err
-	}
-
-	clearedUserVoteList := &OngoingVotedListV2{
-		Direct:               make([]uint64, 0),
-		DelegatedWithAddress: make([]*DelegatedWithAddress, 0),
-	}
-
-	for _, delegatedWithAddress := range userVoteList.DelegatedWithAddress {
-		if delegatedWithAddress.Nonce > lastEndedNonce {
-			clearedUserVoteList.DelegatedWithAddress = append(clearedUserVoteList.DelegatedWithAddress, delegatedWithAddress)
+func (g *governanceContract) clearUserVotesForList(userVoteList *OngoingVotedListV2, lastEndedNonce uint64) {
+	filteredDirect := make([]uint64, 0)
+	for _, direct := range userVoteList.Direct {
```

Comment:
L767-L771 could have been a function that could have been reused also for the delegated list